src/main/java/com/gmail/val59000mc/scenarios/scenariolisteners/MonstersIncListener.java

```diff
@@ -0,0 +1,83 @@
+package com.gmail.val59000mc.scenarios.scenariolisteners;
+
+import com.gmail.val59000mc.UhcCore;
+import com.gmail.val59000mc.scenarios.ScenarioListener;
+
+import java.util.List;
+import java.util.HashMap;
+
+import org.bukkit.Location;
+import org.bukkit.entity.Player;
+import org.bukkit.event.player.PlayerInteractEvent;
+import org.bukkit.block.Block;
+import org.bukkit.block.BlockFace;
+import org.bukkit.block.data.Bisected;
+import org.bukkit.event.block.Action;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.block.BlockPlaceEvent;
+import org.bukkit.event.block.BlockBreakEvent;
+import org.bukkit.metadata.FixedMetadataValue;
+import org.bukkit.metadata.MetadataValue;
+
+public class MonstersIncListener extends ScenarioListener {
+
+    UhcCore plugin = UhcCore.pl;
+    private int DoorsPlaced;
+    private HashMap<Integer, Location> DoorLocs = new HashMap<>();
```

Comment:
I touched on this as part of Consideration #5 - 
"I used a HashMap instead of an ArrayList so that doors could potentially be breakable and removed from the teleport pool in the future." 

I was thinking about allowing doors to be broken and removing them from the pool. Perhaps I am missing something with the way a list works, but I just pictured a map in my head when I was thinking about the possibility of needing to remove a door. I can change this if absolutely needed.

Reply:
You can do that with a list as well.
`list.remove(door);`

Reply:
Hashmap converted to list as of6940a1a